pkg/storegateway/bucket.go

```diff
@@ -349,16 +346,17 @@ func (s *BucketStore) syncBlocks(ctx context.Context, initialSync bool) error {
 		return metaFetchErr
 	}
 
-	// Drop all blocks that are no longer present in the bucket.
-	for id := range s.blocks {
+	s.blocks.Range(func(key, _ any) bool {
```

Comment:
It looks like we were previously iterating over `s.blocks` without locking it and then modifying it at the same time via `s.removeBlock()`? This seems like an improvement :sweat_smile: